### ISSUE TITLE
OS X: Error launching when built with `--asar`

### ISSUE THREAD

**Issue by antifuchs** [zulip/zulip-electron#81]
On OS X, running the output of the default `npm run build:osx` results in a Zulip.app that starts up to this error message, and refuses to continue:

```
Uncaught Exception:
Error: ENOENT, node_modules/simple-spellchecker/dict/en-US.dic not found in /Users/asf/Hacks/zulip-electron/dist/Zulip-darwin-x64/Zulip.app/Contents/Resources/app.asar
    at notFoundError (ELECTRON_ASAR.js:110:19)
    at Object.module.(anonymous function) (ELECTRON_ASAR.js:210:16)
    at OpenReq.process (/Users/asf/Hacks/zulip-electron/dist/Zulip-darwin-x64/Zulip.app/Contents/Resources/app.asar/node_modules/fstream/node_modules/graceful-fs/graceful-fs.js:53:16)
    at OpenReq.Req (/Users/asf/Hacks/zulip-electron/dist/Zulip-darwin-x64/Zulip.app/Contents/Resources/app.asar/node_modules/fstream/node_modules/graceful-fs/graceful-fs.js:124:8)
    at new OpenReq (/Users/asf/Hacks/zulip-electron/dist/Zulip-darwin-x64/Zulip.app/Contents/Resources/app.asar/node_modules/fstream/node_modules/graceful-fs/graceful-fs.js:47:7)
    at Object.open (/Users/asf/Hacks/zulip-electron/dist/Zulip-darwin-x64/Zulip.app/Contents/Resources/app.asar/node_modules/fstream/node_modules/graceful-fs/graceful-fs.js:39:3)
    at WriteStream.open (original-fs.js:2094:6)
    at new WriteStream (original-fs.js:2080:10)
    at Object.fs.createWriteStream (original-fs.js:2035:10)
    at FileWriter._create (/Users/asf/Hacks/zulip-electron/dist/Zulip-darwin-x64/Zulip.app/Contents/Resources/app.asar/node_modules/fstream/lib/file-writer.js:36:19)
```

If I adjust the build command line to drop the `--asar` flag, Zulip starts correctly:

`./node_modules/.bin/electron-packager . --out=dist --app-version=$npm_package_version --prune --icon=app/resources/Icon.icns --overwrite --platform=darwin --arch=x64` results in a Zulip.app that contains those dictionary files.

I suspect this may be a problem with electron-packager, or some asar defaults. Either way, I suspect we might want to leave that flag off the build for now.

I'm testing this on latest zulip-electron `master` HEAD, OS X 10.11.6, using npm 3.10.8 and node v6.9.1.

**Comment by antifuchs** [comment]
A-ha! So an `asar list app.asar` reveals that the en-US dictionary is included, but it lives under a different file name: https://gist.github.com/antifuchs/053004619431e83498bc469cbbedbd6e#file-asar-list-app-asar-L6202 - it's `.zip`, not `.dic`.

I am not sure what's exactly happening but it looks like something is trying to auto-expand that dictionary zip file, and trying to write the expanded .dic back to the asar bundle, which fails. Should the build process just unpack .dic file before building?

**Comment by akashnimare** [comment]
@antifuchs I just tested `npm run build:osx` and it's working fine for me. There are no errors while running Zulip.app. Could you please try this on master -
```sh
$ rm -rf node_modules && npm install

**Comment by antifuchs** [comment]
I should have mentioned that I did this before reporting... no use - on dda491938ec833842a82f1d8abf844de7cd36bc7, I still get this error box. Here's the dependency tree output in a gist: https://gist.github.com/antifuchs/4eea83815bd3bf58f7919d711daa0d57

**Comment by akashnimare** [comment]
@antifuchs yeah, I just did `rm -rf node_modules && npm install` myself and able to reproduce this error. It's an issue with electron-packager's asar api. For now I'm dropping --asar flag to fix this but we should be able to find out the actual reason behind this.

**Comment by akashnimare** [comment]
I think it's caused by this [commit](https://github.com/jfmdev/simple-spellchecker/commit/361266b2c7cb000eee306cb204e8b3014c3a04a1) of simple-spellchecker. I have downgraded it's version to 0.9.0 which is working. Can you check it now? You will have to run `rm -rf node_modules && npm install` on master.

**Comment by akashnimare** [comment]
@antifuchs fixed in [9f70068 ](https://github.com/zulip/zulip-electron/commit/1a63839f6d9b142c73df8bf8a9aac07ca22192dc). Thanks @antifuchs  🙌 .
